### PR TITLE
Fix #55: A11y

### LINKED ISSUE #55: [A11y 8/25] Adjusting slider
With Voiceover/Chrome: not able to use the keyboard to adjust the slider. Also, I'm not able to navigate to the number input text fields to enter a numeric value. 

On Chrome OS/Chromevox, I am able to adjust the slider using the arrow keys. This works well. However, I am not able to access the text to manually input numbers to adjust the value. 

Still need to test with Windows.

### PR DESCRIPTION
**Pull request by blasten**
cc @notwaldorf @frankiefu

### CODE REVIEW

File: paper-slider.html [hunk 1]
```diff
@@ -271,15 +266,15 @@
     },
 
     _minChanged: function() {
-      this.setAttribute('aria-valuemin', this.min);
+      this.$.sliderContainer.setAttribute('aria-valuemin', this.min);
```
Comment (cdata) [inline paper-slider.html]: Does this actually fix the problem? Did you test it with a screen reader? Are there any notable differences between Shadow- and Shadey- DOM?

**Comment by cdata** [comment]
Would it be possible to add some tests for a11y features so that we can describe the desired result and avoid regressions in the future? Also, perhaps we could add `a11ySuite`?

File: paper-slider.html (hunk 1 above)
Comment (blasten) [inline paper-slider.html]: Nope. As far as I can tell it works well in both shady and shadow.

**Comment by blasten** [comment]
@cdata sure! Is there a good example to look at?

**Comment by blasten** [comment]
The test that is failing should pass if https://github.com/PolymerElements/iron-behaviors/pull/30 is merged.

File: paper-slider.html [hunk 2]
```diff
@@ -462,20 +445,29 @@
...
+          this.value = this.min;
+        } else {
+          this.decrement();
+        }
+        this.fire('change');
       }
+    },
+
+    _changeValue: function(event) {
+      this.value = event.detail.value;
```
Comment (frankiefu) [inline paper-slider.html]: I don't think `paper-input`'s `change` event contains `detail.value`

**Comment by frankiefu** [comment]
We need to stop the keydown event propagates from `paper-input` to the host (`paper-slider`), or you will get double increments. e.g. right now if you focus on the input inside the paper-slider and press the `up` arrow key, you will get 2 increments (4 -> 6 -> 8...)

File: paper-slider.html (hunk 2 above)
Comment (blasten) [inline paper-slider.html]: fixed. I'm not sure why I had this. Maybe it used to be?

**Comment by blasten** [comment]
@frankiefu it should be all good now.

**Comment by frankiefu** [comment]
The `up`/`down` arrow keys work fine now.  But what about `left`/`right` arrow keys. Right now when the `paper-input` is in focus and you press `left`/`right` arrow key.  This should move the input cursor, which is expected, but would also cause the slider to increment/decrement the value, which is probably not what we want.

One option is to listen for `keydown` event on `paper-input` and call `stopPropagation`.  What do you think?

File: paper-slider.html [hunk 3]
```diff
@@ -462,20 +445,29 @@
...
+          this.value = this.min;
+        } else {
+          this.decrement();
+        }
+        this.fire('change');
       }
+    },
+
+    _changeValue: function() {
+      this.value = this.$$('#input').value;
```
Comment (frankiefu) [inline paper-slider.html]: You can also just do `event.target.value`

**Comment by frankiefu** [comment]
LGTM :+1:

**Pull request merged**